src/input-select/input-range-select.tsx

```diff
@@ -0,0 +1,316 @@
+import React, { useEffect, useRef, useState } from "react";
+import { ArrowRight, IndicateBar } from "src/date-input/date-input.style";
+import { DropdownList } from "../shared/dropdown-list/dropdown-list";
+import { StringHelper } from "../util/string-helper";
+import { InputSelectWrapper } from "./input-select-wrapper";
+import {
+    Divider,
+    LabelContainer,
+    PlaceholderLabel,
+    RangeIcon,
+    Selector,
+    ValueLabel,
+} from "./input-select.styles";
+import { InputRangeProp, InputRangeSelectProps } from "./types";
+
+type RangeType = "from" | "to";
+
+export const InputRangeSelect = <T, V>({
+    selectedOptions,
+    placeholders = { from: "Select", to: "Select" },
+    options,
+    disabled,
+    readOnly,
+    error,
+    "data-testid": testId,
+    id,
+    enableSearch = false,
+    searchFunction,
+    searchPlaceholder,
+    valueExtractor,
+    valueToStringFunction,
+    listExtractor,
+    displayValueExtractor,
+    onSelectOption,
+    listStyleWidth,
+    onShowOptions,
+    onHideOptions,
+    onRetry,
+    optionsLoadState = { from: "success", to: "success" },
+    optionTruncationType = "middle",
+    renderCustomSelectedOption,
+    renderListItem,
+    renderCustomCallToAction,
+    ...otherProps
+}: InputRangeSelectProps<T, V>): JSX.Element => {
+    // =============================================================================
+    // CONST, STATE
+    // =============================================================================
+    const [selected, setSelected] = useState<InputRangeProp<T>>({
```

Comment:
State objects are hard to manage. Since we're always certain it'll be 2 options, i think we can afford to have 2 state values

Reply:
Done